### PR TITLE
Added fam_progress API to return in flight I/Os

### PR DESCRIPTION
**Pull request by soumya-pn**

### CODE REVIEW

File: examples/api/api_fam_progress.cpp [hunk 1]
```diff
@@ -0,0 +1,130 @@
+/*
+ * api_fam_get_nonblocking.cpp
```
Comment (mashoodk) [inline examples/api/api_fam_progress.cpp:2]: Please correct the filename & copyright

File: examples/api/api_fam_progress.cpp [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    try {
+        // look up the descriptor to a previously allocated data item
+        // (a 50 element integer array)
+        // Fam_Descriptor *descriptor = myFam->fam_lookup("myItem", "myRegion");
+        // allocate local memory to receive 10 elements from different offsets
+        // within the data item
+        int *local1 = (int *)malloc(10 * sizeof(int));
+        int *local2 = (int *)malloc(10 * sizeof(int));
+	uint64_t ret;
```
Comment (mashoodk) [inline examples/api/api_fam_progress.cpp:84]: git-clang-format

File: examples/api/api_fam_progress.cpp [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                                   10 * sizeof(int));
+        // copy elements 26-35 from FAM into local2 elements 0-9 in local memory
+        myFam->fam_get_nonblocking(local2, descriptor, 26 * sizeof(int),
+                                   10 * sizeof(int));
+        int local[] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+        // replace elements 6-15 of the data item in FAM with values in local
+        //          memory
+        myFam->fam_put_nonblocking(local, descriptor, 6 * sizeof(int),
+                                                         10 * sizeof(int));
+        // wait for previously issued non-blocking operations to complete
```
Comment (mashoodk) [inline examples/api/api_fam_progress.cpp:96]: Verify the comments.

File: include/fam/fam.h [hunk 4]
```diff
@@ -1101,6 +1101,13 @@ class fam {
      */
     void fam_quiet(void);
 
+    /**
+     * fam_progress - returns number of all its pending FAM
```
Comment (mashoodk) [inline include/fam/fam.h:1105]:
```suggestion
     * fam_progress - returns number of pending FAM
```

File: test/unit-test/fam-api/fam_progress.cpp [hunk 5]
```diff
@@ -0,0 +1,123 @@
+/*
+ * fam_quiet_put_get_nonblocking.cpp
+ * Copyright (c) 2019 Hewlett Packard Enterprise Development, LP. All rights
```
Comment (mashoodk) [inline test/unit-test/fam-api/fam_progress.cpp:3]: filename and copyright. Please check copyrights in all the files you are touching.

**Review by mashoodk — CHANGES_REQUESTED** [review]

**Review by faizan-barmawer — COMMENTED** [review]
Overall code changes look fine.
Few minor observations:
1. Open a new "documentation" issue to create a man page for this API.
2. Rephrase the commit message title as per Mashood's review comment for the description of API. Add a couple of lines in the commit message to describe the behaviour of the API: return value/error, etc.

Regards, 
Faizan

**Review by mashoodk — APPROVED** [review]

**Review by faizan-barmawer — APPROVED** [review]
Code looks good to merge.

Regards,
Faizan

**Pull request merged**